examples/controls_example/public/app/react_control_example.tsx

```diff
@@ -282,6 +288,33 @@ export const ReactControlExample = ({
             Serialize control group
           </EuiButton>
         </EuiFlexItem>
+        {controlGroupApi && (
+          <EuiFlexItem grow={false}>
+            <EuiButton
+              onClick={async () => {
+                const { type: controlType, state: initialState } = await openDataControlEditor({
```

Comment:
openDataControlEditor promise only resolves when control is saved. Otherwise, the flyout closes and the promise is unresolved. Does this result in a memory leak? How about just passing `controlGroupApi.addNewPanel` to `openDataControlEditor` and avoid the entire promise and async await all together?

Reply:
Good catch! We use `openDataControlEditor` for both editing and creating new controls, so we need access to `replacePanel` as well. And when adding editing a control, we need access to the state manager, which we would have to pass in as... optional? Hmmm...

I think we have two options: 
1. Reject on cancel/close and ignore the rejection - that is how the image embeddable + links embeddable handle it... This is what I was originally going for, I just forget to reject 😆 
2. We pass in the `onSave` as a callback

I've implemented #2 in 2ab084599f96ab62bc7be321552d1ecd5df4bb77 - let me know what you think :+!:

Reply:
+1 to option 2 and passing `onSave` callback. Its much easier to follow and less error prone. Thanks for fixing